j5/backends/hardware/sr/v4/serial/power_board.py

```diff
@@ -0,0 +1,298 @@
+
+        :param identifier: power output to fetch status of.
+        :returns: status of the power output.
+        :raises CommunicationError: Invalid response received.
+        :raises ValueError: Invalid power output identifier.
+        """
+        if identifier in range(6):
+            response = self.query(f"OUT:{identifier}:GET?")
+            if response == "0":
+                return False
+            elif response == "1":
+                return True
+            else:
+                raise CommunicationError(
+                    f"Power Board returned an invalid response: {response}")
+        else:
+            raise ValueError(f"{identifier!r} is not a valid power output identifier")
+
+    def set_power_output_enabled(
+        self, identifier: int, enabled: bool,
+    ) -> None:
+        """
+        Set whether a power output is enabled.
+
+        :param identifier: power output to enable / disable
+        :param enabled: status of the power output.
+        :raises ValueError: Invalid power output identifier.
+        """
+        if identifier in range(6):
+            state = "1" if enabled else "0"
+            self.request(f"OUT:{identifier}:SET:{state}")
+        else:
+            raise ValueError(f"{identifier!r} is not a valid power output identifier")
+
+    def get_power_output_current(self, identifier: int) -> float:
+        """
+        Get the current being drawn on a power output, in amperes.
+
+        :param identifier: power output to fetch current of.
+        :returns: current of the output.
+        :raises ValueError: Invalid power output identifier.
+        """
+        if identifier in range(6):
+            response = self.query(f"OUT:{identifier}:I?")
+            return float(response) / 1000
+        else:
+            raise ValueError(f"{identifier!r} is not a valid power output identifier")
+
+    def buzz(
+        self,
+        identifier: int,
+        duration: timedelta,
+        frequency: float,
+        blocking: bool,
+    ) -> None:
+        """
+        Queue a pitch to be played.
+
+        :param identifier: piezo identifier to play pitch on.
+        :param duration: duration of the tone.
+        :param frequency: Pitch of the tone in Hz.
+        :param blocking: whether the code waits for the buzz
+        :raises ValueError: invalid value for parameter.
+        :raises NotSupportedByHardwareError: unsupported pitch freq or length.
+        """
+        if identifier != 0:
+            raise ValueError(f"Invalid piezo identifier {identifier!r}; "
+                             f"the only valid identifier is 0.")
+
+        duration_ms = round(duration.total_seconds() * 1000)
+
+        if not 0 < duration_ms < MAX_BUZZ_DURATION_MS:
+            raise NotSupportedByHardwareError(
+                f"Piezo duration must be in range of 0 - {MAX_BUZZ_DURATION_MS}ms.",
+            )
+
+        frequency_int = int(round(frequency))
+        if not 0 < frequency_int < 10_000:
```

Comment:
This range excludes 0 which would be sent to silence the buzzer.

Reply:
0 isn't a valid value in the j5 component spec anyway.

https://github.com/j5api/j5/blob/main/j5/components/piezo.py#L145-L146

Tracking this in #820

Reply:
This seems like a substantial shortcoming given that sending 0 frequency buzzes was how the old firmware did gaps in tones.

Reply:
This will have to be addressed later down the line. We don't have time at the moment and I'm afraid I don't agree that this is substantial.